### ISSUE TITLE
How to change the icon?

### ISSUE THREAD

**Issue by ChrisLMerrill** [edvin/fxlauncher#41]
I could not find anything in the fxlauncher docs about changing the icon of either the installer or the installed program. Is there a way?

**Comment by edvin** [comment]
The icon can be changed with the `icon=path` parameter to javapackager, as described in the official docs:

http://docs.oracle.com/javase/8/docs/technotes/tools/unix/javapackager.html

For a Maven project that would mean adding the following argument to the installer/javapackager step:

```xml
<argument>-Bicon=src/main/resources/cube.ico</argument>
```

**Comment by ChrisLMerrill** [comment]
Sorry...I should have been more specific - I'm using Gradle. I can see how the params get passed in Maven example, but I don't see how that would happen in the Gradle example. I guess I need to dig into your source to answer that.

**Comment by ChrisLMerrill** [comment]
Looking at the source, it doesn't look like the icon parameter is supported (or any way to pass arbitrary parameters). I'll try to find time to build a fix for that.

**Comment by edvin** [comment]
That would be great. It should be pretty straight forward. I think it would be nice to add specific support for the icon parameter, as well as a way to add arbitrary parameters. If you find the time to do so I'd be most grateful, and please ask if you get stuck in any way.

**Comment by ChrisLMerrill** [comment]
So I kinda went back-and-forth on this. The other project I'd tried for building native installers has a lot more configuration options (javafx-gradle-plugin), including icon support. But I wanted to keep the auto-updating capabilities. So I decided to try both. I used javafx-gradle-plugin in a separate project to build a native installer. That is calling FxLauncher with the -app parameter to update and launch my app. Worked great! Eliminates the need to enhance FxLauncher with icon support.

**Comment by edvin** [comment]
OK, thanks for getting back to me. I might add it myself anyway :)

**Comment by arthurcvm** [comment]
In what block do I use this in pom.xml?

> For a Maven project that would mean adding the following argument to the installer/javapackager step (example):

`<argument>-Bicon=src/main/resources/icon.ico</argument>`

**Comment by Koire** [comment]
@ChrisLMerrill 
Any chance you can share that set up with me? I'm actually working on the exact same thing right now

**Comment by edvin** [comment]
The icon can actually be specified with the `javapackagerOptions` parameter, which accepts a `List<String>` :)

**Comment by ChrisLMerrill** [comment]
@MrVlad001 I have one project that creates the native launcher using javafx-gradle-plugin. It has a single class with a main. It then calls into Launcher.main from fxlauncher to start my app. A second project creates my resources for fxlauncher. I haven't touched the native launcher (1st project) since Dec 2016. I only need to change the app resources at the fxlauncher app URL. 

When I did this, it gave me a number of advantages:
1. Use javafx-gradle-plugin for building the launcher, which had more options than fxlauncher (I don't recall which one of those I really wanted, but there was one that sent me in this direction).
2. Use fxlauncher for automatic updates (love this!)
3. I created a signed installer for Windows and haven't had to do that again because fxlauncher updates to the latest on the first run.
4. My class that is the middle-man between the native launcher and fxlauncer gives me the opportunity to control the launch a bit - which allowed me to implement separate release channels. Users are on the stable channel by default. But an advanced user can create 2 config files in their profile which will change the local folder and fxlauncher app URL that are passed to fxlauncher. This switches the user to a development channel to get bleeding-edge versions as I release them.

Hope that helps!
Chris

**Comment by MrVlad001** [comment]
Many thanks @ChrisLMerrill ,
Could you show me how you call your app from the first app?
Something like this ?
`Application.launch(Launcher.class,"--app=http://remote/location/app.xml");`

**Comment by simplesteph** [comment]
@MrVlad001  did you manage to make that work? That is crashing for me by trying to seek the file locally

@ChrisLMerrill  the sample setup would be greatly appreciated, I'd love to reproduce what you have

**Comment by ChrisLMerrill** [comment]
Sorry to both of you for not responding last year. I ended up moving away from fxlauncher with javafx-gradle-plugin due to a need for Java 10 support (now Java 11 and 12). I had separated development of the installer from the updater/launcher, anyway.

For the installer, I'm just using javapackger, from this gist: https://gist.github.com/falkoschumann/6c0abf0f9f65a02babad6b070dcf4598
For launch/update, I moved to update4j (https://github.com/update4j/update4j)

I ended up writing my own Gradle plugin to work with update4j (https://github.com/ChrisLMerrill/update4j-gradle-plugin). That was an adventure :)  

I'm sure that is not the answer you wanted to hear...I doubt I could dig up my old code that was using fxlauncher.
